CMSgov/easi-app
# [EASI-4007] initial pass at connecting TRB Requests to contracts/systems/services - base Graph PR

# EASI-4007


## Changes and Description

- DB changes:
  - Add column to `trb_request` table: `contract_name ZERO_STRING`
  - Create enum `trb_request_relation_type` with options: `'NEW_SYSTEM', 'EXISTING_SYSTEM', 'EXISTING_SERVICE'`
  - Add column to `trb_request` table: `system_relation_type` of the above enum type `trb_request_relation_type`
- Graph changes:
  - Create the following mutations (and accompanying input types):
    - `setTRBRequestRelationNewSystem(input: SetTRBRequestRelationNewSystemInput): TRBRequest`
    - `setTRBRequestRelationExistingSystem(input: SetTRBRequestRelationExistingSystemInput): TRBRequest`
    - `setTRBRequestRelationExistingService(input: SetTRBRequestRelationExistingServiceInput): TRBRequest`
    - `unlinkTRBRequestRelation(trbRequestID: UUID!): TRBRequest`
    - centralize enum regarding `RequestRelationType`
- code changes:
  - create resolvers for the above four new graph mutations
  - write tests for those resolvers
  - centralize code regarding `RequestRelationType`

  



## How to test this change
- confirm tests pass
- through Postman, you can create a TRB Request, then set relation types in the `Set TRB Relations` folder



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
